[DOC] Add CLI section to user documentation. (#12802)
Scille/parsec-cloud: docs/cli/device.rst

.. _doc_cli_device:

===========================

> Device related CLI commands

Reviewer: I would shorten these section titles
```suggestion
Device commands
```

===========================

> In this context, a device is not a physical device but the encrypted key to access a parsec organization as a given user.

Reviewer: You can reference to existing docs:
https://docs.parsec.cloud/en/latest/hosting/architecture.html#key-parsec-concepts

.. Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS

.. _doc_cli_device:

===================
Device CLI commands
===================

In this context, a device is not a physical device but the encrypted key to access a parsec organization as a given user.
Usually one user will have one device per organization per physical device.

> For more see https://docs.parsec.cloud/en/latest/hosting/architecture.html#key-parsec-concepts

Reviewer: I wouldn't rephrase what's already described in the Architecture section
```suggestion
In this context, a device is not a physical device but the encrypted file containing the keys allowing the User to access the Organization.

See :ref:`Key Parsec concepts <doc_hosting_architecture_concepts>`
```
Remember to add the corresponding anchor in architecture.rst

```rst
.. _doc_hosting_architecture_concepts:
```

List
----

List all devices available locally. The information provided for each device is:

- short id
- organization name
- user name
- email
- device label
- parsec server address

.. code-block:: shell

    parsec-cli device list
    Found 4 device(s) in /tmp/parsec-testenv-831f316f-5fa7-41f4-a7f8-ba5c5658d79b/config/parsec3/libparsec:
    052 - Org: Alice <alice@example.com> @ pc (parsec3://127.0.0.1:7778?no_ssl=true)
    32c - Org: Bob <bob@example.com> @ laptop (parsec3://127.0.0.1:7778?no_ssl=true)
    b1e - Org: Alice <alice@example.com> @ laptop (parsec3://127.0.0.1:7778?no_ssl=true)
    c90 - Org: Toto <toto@example.com> @ laptop (parsec3://127.0.0.1:7778?no_ssl=true)

.. note::

    In this example the listed devices are all for the same organization but for different users,

>     this is because this command was ran in a test environment.

Reviewer: I would remove this note and tweak the output to a more "real" case.

Typically, the user will have devices for its own user (and eventually, different organizations).

.. _cli_device_forget_local:

Forget-local
------------

This command deletes locally the selected device. This does not involve any server operation.

> If the device file has been saved elsewhere, using that device file will grant access to the server.

Reviewer: Are you trying to warn the reader about it? 
If that's the case, I would put it in a warning or important admonition.
```suggestion

.. important:

  Please note that if the device file exists elsewhere, it can still be used to
  access the organization on this server. In other words, this command does not 
  replace user revocation in case of a compromised device.
```

.. code-block:: shell

    parsec-cli device forget-local -d c90
    You are about to forget the following local device:
    c90 - Org: Toto <toto@example.com> @ laptop
    Are you sure? yes
    The local device has been forgotten

Change authentication
---------------------

This command changes the authentication method to password or keyring.

> More authentication method are supported by the GUI.

Reviewer: ```suggestion
Other authentication methods are supported by the GUI.
```

.. code-block:: shell

    parsec-cli device change-authentication -d 052 --password
    Enter current password for the device:
    Enter new password for the device:
    Confirm password:
    Device authentication changed successfully

.. note::

    In this example, the authentication method has not changed per se, but

>     the password itself have changed.

Reviewer: ```suggestion
    the password itself has been updated.
```

> .. _doc_cli_device:

Author: add copyright
